feat(cargo): support private crate (#24704)
renovatebot/renovate: docs/usage/rust.md

## Private crate registries and private Git dependencies

> This example shows how you can use a `hostRules` configuration to configure Renovate for use with a custom private Crate using Git to pull the modules when updating `Cargo.lock`.

Reviewer: This sentence does too much at once:

- Thing about `hostRules`
- Thing about custom private Crate using Git
- Pull modules when updating `Cargo.lock`

I can't keep all this in my head. So we should split the sentence into multiple parts.

> All token `hostRules` with a `hostType` (e.g. `github`, `gitlab`, `bitbucket`, ... ) and host rules without a `hostType` are setup for authentication.

Reviewer: > are setup for authentication.

This is passive. I think we mean: "You as user can set authentication by doing x."

> It's also possible to setup a hostRules which will be used only for Cargo authentication (e.g. `hostType: 'cargo'`).

Reviewer: ```suggestion
You can configure a `hostRules` that's only for Cargo authentication (e.g. `hostType: 'cargo'`).
```

Let's be clear on _who_ does what, and simplify a bit.
